### PR TITLE
Algolia and Notifications

### PR DESCRIPTION
**Pull request by noahtallen**
Activities and Users sync to Algolia.
On activity creation, send notification to nearby users.

### CODE REVIEW

File: functions/src/notifications.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    activityId: string,
+    ownerId: string,
+    topicIds: string[],
+    coords: FirebaseFirestore.GeoPoint,
+}
+
+export const onActivityCreation = firestore.document('activities/{activityId}').onCreate(activityCreationHandler)
+
+export async function activityCreationHandler(snap: FirebaseFirestore.DocumentSnapshot, context: EventContext) {
+    const newData = exports.getActivityData(snap)
```
Comment (noahtallen) [inline functions/src/notifications.ts]: Here, we call `exports` so that it's the same thing that we're spying on in tests

File: functions/src/test/notifications.test.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+// NOTE: The `@ts-ignore` statements in this file are because much of the data
+// does not fit the shape of the expected data. This is what we want, because
+// we're trying to test the edge cases where the data doesn't fit what's expected.
+
+describe('Activity creation handler', () => {
+    it('Gets activity data based on the data changed', async () => {
+        const dataSpy = sinon.spy(notifications, 'getActivityData')
+        // @ts-ignore
+        await notifications.activityCreationHandler(mocks.testActivitySnap, {})
+        assert(dataSpy.calledOnce)
```
Comment (noahtallen) [inline functions/src/test/notifications.test.ts]: Here we spy on the export of the other file to make sure it was called by the function

File: functions/src/notifications.ts (hunk 1 above)
Comment (lukemeierj) [reply functions/src/notifications.ts]: Ahhh! This is the thing I didn't do!  How'd you figure that out?

Mocking is sooo hard when you don't have a great grasp of the module management of a language.

**Comment by noahtallen** [comment]
This got too busy, making a new one
